### PR TITLE
[GNA] Fix of unsupported size of Diagonal layer

### PR DESCRIPTION
**Pull request by elilobanova**

### CODE REVIEW

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp [hunk 1]
```diff
@@ -286,7 +286,7 @@ inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
         THROW_IE_EXCEPTION << "Unsupported input dims size for " << wl->name << ", should be > 1, but " << wl->insData[0].lock().get()->getDims().size();
     }
     uint32_t num_rows = isDiagonal ? 1 : wl->outData[0]->getDims()[oIdx];
-    uint32_t num_columns = wl->insData[0].lock().get()->getDims()[iIdx];
+    uint32_t num_columns = isDiagonal ? wl->_weights->size() : wl->insData[0].lock().get()->getDims()[iIdx];
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:289]: In diagonal case, number of input columns should be equal to weights size.
If there is mismatch, then most likely it is bug in GNAGraphCompiler::AffinePrimitive

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 2]
```diff
@@ -522,23 +522,12 @@ void GNAGraphCompiler::PowerPrimitive(InferenceEngine::CNNLayerPtr layer) {
...
     auto outputs = *layer->outData.begin();
-    uint32_t num_rows_in = InferenceEngine::details::product(begin(input->getDims()), end(input->getDims()));
-    uint32_t num_columns_in = 1;
+    auto reshaped_dims = Get2DReshapedData(input, 8)->getDims();
+    uint32_t num_rows_in = reshaped_dims[1];
+    uint32_t num_columns_in = reshaped_dims[0];
     uint32_t num_rows_out = num_rows_in;
     uint32_t num_padding = ALIGN(num_rows_in, 8) - num_rows_in;
 
-    if (input->getDims().size() > 2 || input->getDims()[0] >= 8) {
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:530]: @admitriev-gna added it for power broadcast case. Why is this removed?

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp (hunk 1 above)
Comment (elilobanova) [reply inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:289]: In this fix input data for a diagonal layer is reshaped in GNAGraphCompiler::AffinePrimitive to have batch size to be equal to some value from 8 to 1.  wl->insData[0] contains dims before this reshaping, at the same time size of weights created in Pass Manager is equal to input columns number after reshape. So here we need to use weights size which corresponds to actually created weights.

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp (hunk 2 above)
Comment (elilobanova) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:530]: This code is moved to Get2DReshapedData() function which must be general for Power and Diagonal primitives.

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp (hunk 1 above)
Comment (bsochack) [reply inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:289]: thanks for detailed explanation.

**Review by bsochack — APPROVED** [review]
